compiler/rustc_codegen_ssa/src/codegen_attrs.rs

```diff
@@ -327,6 +327,10 @@ fn codegen_fn_attrs(tcx: TyCtxt<'_>, did: LocalDefId) -> CodegenFnAttrs {
                     } else {
                         codegen_fn_attrs.linkage = linkage;
                     }
+                    if tcx.is_mutable_static(did.into()) {
+                        tcx.dcx()
+                            .span_err(attr.span, "mutable statics are not allowed with #[linkage]");
```

Comment:
```suggestion
                            .span_err(attr.span, "mutable statics are not allowed with `#[linkage]`");
```

Also maybe worth while explaining *why* in a `note`, or linking to something that explains why.

Reply:
Done